File: 136_single_number/hash.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        unordered_map<int, int>occur;
        for(int i = 0; i < nums.size(); ++i){
            auto it = occur.find(nums[i]);
            if(it != occur.end()) it->second += 1;
            else occur.insert({nums[i], 1});
        }
        for(auto it = occur.begin();it != occur.end(); ++it){
            if(it->second == 1) return it->first;
        }
        return 0;
    }
};

File: 136_single_number/hash.py
class Solution:
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        """
        Complexity: O(n)
        Space: O(n)
        """
        occur = {}
        for i in range(len(nums)):
            if nums[i] not in occur:
                occur[nums[i]] = 1
            else:
                occur[nums[i]] += 1
        for k in occur:
            if occur[k] == 1:
                return k
        return 0
